### ISSUE TITLE
MVStore driver stuck at "Starting"

### ISSUE THREAD

**Issue by nicolas-f** [opensensorhub/osh-persistence#3]
When I click "start" on H2 MvStore storage driver the state is stuck on "Starting" without displaying any data.

Here the jstack if it helps

```
Full thread dump Java HotSpot(TM) Client VM (25.162-b12 mixed mode):

"Atmosphere-Scheduler-45" #205 daemon prio=5 os_prio=0 tid=0x53406800 nid=0x77ad waiting on condition [0x5097e000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x5eea4760> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.poll(ScheduledThreadPoolExecutor.java:1129)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.poll(ScheduledThreadPoolExecutor.java:809)
        at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1073)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

"Timer-12" #177 prio=5 os_prio=0 tid=0x5114a400 nid=0x6772 in Object.wait() [0x50eae000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        - waiting on <0x5f45db00> (a java.util.TaskQueue)
        at java.lang.Object.wait(Object.java:502)
        at java.util.TimerThread.mainLoop(Timer.java:526)
        - locked <0x5f45db00> (a java.util.TaskQueue)
        at java.util.TimerThread.run(Timer.java:505)

"Attach Listener" #169 daemon prio=9 os_prio=0 tid=0x52e5b000 nid=0x62c8 waiting on condition [0x00000000]
   java.lang.Thread.State: RUNNABLE

"Timer-11" #163 prio=5 os_prio=0 tid=0x52cbe800 nid=0x61df in Object.wait() [0x509ce000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        - waiting on <0x5f40fd00> (a java.util.TaskQueue)
        at java.lang.Object.wait(Object.java:502)
        at java.util.TimerThread.mainLoop(Timer.java:526)
        - locked <0x5f40fd00> (a java.util.TaskQueue)
        at java.util.TimerThread.run(Timer.java:505)

"Timer-9" #120 prio=5 os_prio=0 tid=0x52e39800 nid=0x5f4a in Object.wait() [0x50abe000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        - waiting on <0x5f2ba5c8> (a java.util.TaskQueue)
        at java.lang.Object.wait(Object.java:502)
        at java.util.TimerThread.mainLoop(Timer.java:526)
        - locked <0x5f2ba5c8> (a java.util.TaskQueue)
        at java.util.TimerThread.run(Timer.java:505)

"Scheduler-10865127" #35 prio=5 os_prio=0 tid=0x0014a400 nid=0x55b9 waiting on condition [0x513b1000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x5ee17a48> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1093)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:809)
        at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

"Atmosphere-Scheduler-0" #32 daemon prio=5 os_prio=0 tid=0x521cf800 nid=0x55b6 waiting on condition [0x514a1000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x5eea4760> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.poll(ScheduledThreadPoolExecutor.java:1134)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.poll(ScheduledThreadPoolExecutor.java:809)
        at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1073)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

"Scheduler-17045491" #30 prio=5 os_prio=0 tid=0x52ac6400 nid=0x558d waiting on condition [0x5170d000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x5edfb9b0> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1093)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:809)
        at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

"qtp20964191-29" #29 prio=5 os_prio=0 tid=0x52abb400 nid=0x558c waiting on condition [0x5175d000]
   java.lang.Thread.State: WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x5ef29060> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.reservedWait(ReservedThreadExecutor.java:193)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:225)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:679)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:597)
        at java.lang.Thread.run(Thread.java:748)

"qtp20964191-28" #28 prio=5 os_prio=0 tid=0x52ab9c00 nid=0x558b runnable [0x517ad000]
   java.lang.Thread.State: RUNNABLE
        at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
        at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:269)
        at sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:93)
        at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
        - locked <0x5edfc780> (a sun.nio.ch.Util$3)
        - locked <0x5edfc790> (a java.util.Collections$UnmodifiableSet)
        - locked <0x5edfc740> (a sun.nio.ch.EPollSelectorImpl)
        at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
        at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:101)
        at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.select(ManagedSelector.java:346)
        at org.eclipse.jetty.io.ManagedSelector$SelectorProducer.produce(ManagedSelector.java:299)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:179)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:140)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:243)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:679)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:597)
        at java.lang.Thread.run(Thread.java:748)

"qtp20964191-27" #27 prio=5 os_prio=0 tid=0x52ab8400 nid=0x558a waiting on condition [0x517fc000]
   java.lang.Thread.State: WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x5f49f568> (a java.util.concurrent.CountDownLatch$Sync)
        at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:997)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304)
        at java.util.concurrent.CountDownLatch.await(CountDownLatch.java:231)
        at org.atmosphere.container.BlockingIOCometSupport.suspend(BlockingIOCometSupport.java:147)
        at org.atmosphere.container.BlockingIOCometSupport.service(BlockingIOCometSupport.java:65)
        at org.atmosphere.cpr.AtmosphereFramework.doCometSupport(AtmosphereFramework.java:2075)
        at com.vaadin.server.communication.PushRequestHandler.handleRequest(PushRequestHandler.java:248)
        at com.vaadin.server.VaadinService.handleRequest(VaadinService.java:1422)
        at com.vaadin.server.VaadinServlet.service(VaadinServlet.java:380)
        at org.sensorhub.ui.AdminUIServlet.service(AdminUIServlet.java:51)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:841)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
        at org.eclipse.jetty.servlets.CrossOriginFilter.handle(CrossOriginFilter.java:308)
        at org.eclipse.jetty.servlets.CrossOriginFilter.doFilter(CrossOriginFilter.java:262)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:513)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
        at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:61)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.eclipse.jetty.server.Server.handle(Server.java:561)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:334)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251)
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:104)
        at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:247)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:140)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:243)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:679)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:597)
        at java.lang.Thread.run(Thread.java:748)

"qtp20964191-26" #26 prio=5 os_prio=0 tid=0x52ab6c00 nid=0x5589 waiting on condition [0x5184d000]
   java.lang.Thread.State: WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x5eed6720> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.reservedWait(ReservedThreadExecutor.java:193)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:225)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:679)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:597)
        at java.lang.Thread.run(Thread.java:748)

"qtp20964191-25" #25 prio=5 os_prio=0 tid=0x52ab5400 nid=0x5588 waiting on condition [0x5189d000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x5ed1bce0> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
        at org.eclipse.jetty.util.BlockingArrayQueue.poll(BlockingArrayQueue.java:392)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.idleJobPoll(QueuedThreadPool.java:571)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.access$800(QueuedThreadPool.java:50)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:634)
        at java.lang.Thread.run(Thread.java:748)

"qtp20964191-24" #24 prio=5 os_prio=0 tid=0x52ab4400 nid=0x5587 waiting on condition [0x518ed000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x5ed1bce0> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
        at org.eclipse.jetty.util.BlockingArrayQueue.poll(BlockingArrayQueue.java:392)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.idleJobPoll(QueuedThreadPool.java:571)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.access$800(QueuedThreadPool.java:50)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:634)
        at java.lang.Thread.run(Thread.java:748)

"qtp20964191-23-acceptor-0@39327e-ServerConnector@fedac9{HTTP/1.1,[http/1.1]}{0.0.0.0:8181}" #23 prio=3 os_prio=0 tid=0x52aaf000 nid=0x5586 runnable [0x5193d000]
   java.lang.Thread.State: RUNNABLE
        at sun.nio.ch.ServerSocketChannelImpl.accept0(Native Method)
        at sun.nio.ch.ServerSocketChannelImpl.accept(ServerSocketChannelImpl.java:422)
        at sun.nio.ch.ServerSocketChannelImpl.accept(ServerSocketChannelImpl.java:250)
        - locked <0x5edfc550> (a java.lang.Object)
        at org.eclipse.jetty.server.ServerConnector.accept(ServerConnector.java:379)
        at org.eclipse.jetty.server.AbstractConnector$Acceptor.run(AbstractConnector.java:643)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:679)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:597)
        at java.lang.Thread.run(Thread.java:748)

"qtp20964191-22" #22 prio=5 os_prio=0 tid=0x52aae400 nid=0x5585 waiting on condition [0x51dcd000]
   java.lang.Thread.State: WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x5efcc560> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.reservedWait(ReservedThreadExecutor.java:193)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:225)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:679)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:597)
        at java.lang.Thread.run(Thread.java:748)

"Timer-2" #20 prio=5 os_prio=0 tid=0x51f28000 nid=0x5583 in Object.wait() [0x51cdd000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        - waiting on <0x5ed76068> (a java.util.TaskQueue)
        at java.util.TimerThread.mainLoop(Timer.java:552)
        - locked <0x5ed76068> (a java.util.TaskQueue)
        at java.util.TimerThread.run(Timer.java:505)

"Timer-1" #19 prio=5 os_prio=0 tid=0x51f26800 nid=0x5582 in Object.wait() [0x51d2d000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        - waiting on <0x5ed76228> (a java.util.TaskQueue)
        at java.util.TimerThread.mainLoop(Timer.java:552)
        - locked <0x5ed76228> (a java.util.TaskQueue)
        at java.util.TimerThread.run(Timer.java:505)

"Timer-0" #18 prio=5 os_prio=0 tid=0x51f23400 nid=0x5581 in Object.wait() [0x51d7d000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        - waiting on <0x5ed763e8> (a java.util.TaskQueue)
        at java.util.TimerThread.mainLoop(Timer.java:552)
        - locked <0x5ed763e8> (a java.util.TaskQueue)
        at java.util.TimerThread.run(Timer.java:505)

"DestroyJavaVM" #16 prio=5 os_prio=0 tid=0x76209000 nid=0x5546 waiting on condition [0x00000000]
   java.lang.Thread.State: RUNNABLE

"Service Thread" #7 daemon prio=9 os_prio=0 tid=0x52f9b800 nid=0x554f runnable [0x00000000]
   java.lang.Thread.State: RUNNABLE

"C1 CompilerThread0" #6 daemon prio=9 os_prio=0 tid=0x52f8c800 nid=0x554e waiting on condition [0x00000000]
   java.lang.Thread.State: RUNNABLE

"Signal Dispatcher" #5 daemon prio=9 os_prio=0 tid=0x52f8b400 nid=0x554d runnable [0x00000000]
   java.lang.Thread.State: RUNNABLE

"Finalizer" #3 daemon prio=8 os_prio=0 tid=0x76266800 nid=0x554c in Object.wait() [0x5375e000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        - waiting on <0x5eb6a3e0> (a java.lang.ref.ReferenceQueue$Lock)
        at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:143)
        - locked <0x5eb6a3e0> (a java.lang.ref.ReferenceQueue$Lock)
        at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:164)
        at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:212)

"Reference Handler" #2 daemon prio=10 os_prio=0 tid=0x76263c00 nid=0x554b in Object.wait() [0x537ae000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        - waiting on <0x5eb6a580> (a java.lang.ref.Reference$Lock)
        at java.lang.Object.wait(Object.java:502)
        at java.lang.ref.Reference.tryHandlePending(Reference.java:191)
        - locked <0x5eb6a580> (a java.lang.ref.Reference$Lock)
        at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:153)

"VM Thread" os_prio=0 tid=0x7625f400 nid=0x554a runnable 

"VM Periodic Task Thread" os_prio=0 tid=0x52f9d800 nid=0x5550 waiting on condition 

JNI global references: 227
```

Log say nothing interesting

```
2018-06-13 13:24:24.820 INFO  [ModuleRegistry-9] - Module INITIALIZING
2018-06-13 13:24:24.825 INFO  [ModuleRegistry-9] - Module INITIALIZED
2018-06-13 13:24:33.659 INFO  [ModuleRegistry-9] - Module STARTING
```

But journalctl is more verbose

```
Jun 13 14:24:33 raspberrypi sh[21827]: 2018-06-13 13:24:33.656 INFO  AdminUIModule:125edb9e [qtp20964191-26] - New UI action: Start 'NoiseMonitoringLocal' [e8169fe2-52b2-4b89-b6e7-8e864fbeee03] (from ip=192.168.1.102, user=admin
Jun 13 14:24:33 raspberrypi sh[21827]: 2018-06-13 13:24:33.659 INFO  GenericStreamStorage:896db6b1 [ModuleRegistry-9] - Module STARTING
Jun 13 14:24:33 raspberrypi sh[21827]: 2018-06-13 13:24:33.672 INFO  ModuleRegistry [EventBus-32] - Starting module 'NoiseMonitoringLocal' [e8169fe2-52b2-4b89-b6e7-8e864fbeee03]
Jun 13 14:24:33 raspberrypi sh[21827]: Exception in thread "Timer-10" java.lang.IllegalStateException: Last block not stored, possibly due to out-of-memory [1.4.196/3]
Jun 13 14:24:33 raspberrypi sh[21827]: at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:765)
Jun 13 14:24:33 raspberrypi sh[21827]: at org.h2.mvstore.MVStore.storeNowTry(MVStore.java:1094)
Jun 13 14:24:33 raspberrypi sh[21827]: at org.h2.mvstore.MVStore.storeNow(MVStore.java:1050)
Jun 13 14:24:33 raspberrypi sh[21827]: at org.h2.mvstore.MVStore.commitAndSave(MVStore.java:1039)
Jun 13 14:24:33 raspberrypi sh[21827]: at org.h2.mvstore.MVStore.commit(MVStore.java:1000)
Jun 13 14:24:33 raspberrypi sh[21827]: at org.sensorhub.impl.persistence.h2.MVObsStorageImpl.commit(MVObsStorageImpl.java:180)
Jun 13 14:24:33 raspberrypi sh[21827]: at org.sensorhub.impl.persistence.MaxAgeAutoPurgePolicy.trimStorage(MaxAgeAutoPurgePolicy.java:84)
Jun 13 14:24:33 raspberrypi sh[21827]: at org.sensorhub.impl.persistence.GenericStreamStorage$1.run(GenericStreamStorage.java:126)

```

I've allocated 512 mb to open sensor hub and I have 4.6 Gb available on the device.

I have added in libs

h2-mvstore-1.4.196.jar
sensorhub-storage-h2-1.3.2.jar
kryo-4.0.0.jar

Any idea ?

**Comment by alexrobin** [comment]
Can you reproduce the problem in a different environment than the Raspberry Pi?

**Comment by nicolas-f** [comment]
Yes same issue with the fakeweather driver on my pc. Maybe if you have a version of sensorhub-storage-h2-1.3.2.jar that I can download to test.

**Comment by nicolas-f** [comment]
This is a deadlock at the exit of synchronized block osh-core/sensorhub-core/src/main/java/org/sensorhub/impl/module/AbstractModule.javaAbstractModuleJava:518

**Comment by nicolas-f** [comment]
A class not found exception trigger that deadlock (prevent Started set)
The missing jar was
minlog-1.3.0.jar used by kryo-4.0.0.jar

**Comment by nicolas-f** [comment]
I close it has it is fixed.

**Comment by alexrobin** [comment]
Is the following issue with MVStore still happening?

```
Jun 13 14:24:33 raspberrypi sh[21827]: Exception in thread "Timer-10" java.lang.IllegalStateException: Last block not stored, possibly due to out-of-memory [1.4.196/3]
```

**Comment by nicolas-f** [comment]
It was caused by a thread crash triggered by class not found when de-serialize

**Comment by nicolas-f** [comment]
I think in the next version, 128mb default jvm will not be sufficant:

```
Jun 14 15:14:05 raspberrypi sh[15604]: Exception in thread "EventBus-643" java.lang.OutOfMemoryError: Capacity: 10616832
Jun 14 15:14:05 raspberrypi sh[15604]: at org.h2.mvstore.WriteBuffer.grow(WriteBuffer.java:322)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.h2.mvstore.WriteBuffer.ensureCapacity(WriteBuffer.java:302)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.h2.mvstore.WriteBuffer.put(WriteBuffer.java:177)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.sensorhub.impl.persistence.h2.KryoDataType.write(KryoDataType.java:115)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.sensorhub.impl.persistence.h2.KryoDataType.write(KryoDataType.java:127)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.h2.mvstore.Page.write(Page.java:744)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.h2.mvstore.Page.writeUnsavedRecursive(Page.java:825)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.h2.mvstore.Page.writeUnsavedRecursive(Page.java:831)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.h2.mvstore.Page.writeUnsavedRecursive(Page.java:831)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.h2.mvstore.MVStore.storeNowTry(MVStore.java:1151)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.h2.mvstore.MVStore.storeNow(MVStore.java:1050)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.h2.mvstore.MVStore.commitAndSave(MVStore.java:1039)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.h2.mvstore.MVStore.commit(MVStore.java:1000)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.sensorhub.impl.persistence.h2.MVObsStorageImpl.commit(MVObsStorageImpl.java:180)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.sensorhub.impl.persistence.GenericStreamStorage.handleEvent(GenericStreamStorage.java:462)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.sensorhub.impl.common.AsyncEventHandler$ListenerQueue$1.run(AsyncEventHandler.java:91)
Jun 14 15:14:05 raspberrypi sh[15604]: at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
Jun 14 15:14:05 raspberrypi sh[15604]: at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
Jun 14 15:14:05 raspberrypi sh[15604]: at java.lang.Thread.run(Thread.java:748)
```

And H2 close the map after this.

```
Jun 14 15:14:05 raspberrypi sh[15604]: 2018-06-14 14:14:05.665 ERROR AsyncEventHandler [EventBus-656] - Uncaught exception while dispatching event from FastAcousticOutput to GenericStreamStorage
Jun 14 15:14:05 raspberrypi sh[15604]: java.lang.IllegalStateException: This map is closed [1.4.196/4]
Jun 14 15:14:05 raspberrypi sh[15604]: at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:765)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.h2.mvstore.MVMap.beforeWrite(MVMap.java:1044)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.h2.mvstore.MVMap.put(MVMap.java:122)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.h2.mvstore.MVMap.putIfAbsent(MVMap.java:571)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.sensorhub.impl.persistence.h2.MVTimeSeriesImpl.store(MVTimeSeriesImpl.java:270)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.sensorhub.impl.persistence.h2.MVObsStorageImpl.storeRecord(MVObsStorageImpl.java:505)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.sensorhub.impl.persistence.GenericStreamStorage.handleEvent(GenericStreamStorage.java:416)
Jun 14 15:14:05 raspberrypi sh[15604]: at org.sensorhub.impl.common.AsyncEventHandler$ListenerQueue$1.run(AsyncEventHandler.java:91)
Jun 14 15:14:05 raspberrypi sh[15604]: at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
Jun 14 15:14:05 raspberrypi sh[15604]: at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
Jun 14 15:14:05 raspberrypi sh[15604]: at java.lang.Thread.run(Thread.java:748)
```
